### ISSUE TITLE
Zero length slice casting

### ISSUE THREAD

**Issue by DarkEld3r** [sdroege/byte-slice-cast#6]
The following conversion fails with the `AlignmentMismatch` error:
```rust
let slice: Vec<u8> = Vec::new();
let converted_slice = slice.as_slice_of::<u16>().unwrap();
```
It is more a question than an issue: can such conversions be made? It will make writing generic code easier.

**Comment by sdroege** [comment]
Thanks, fixing :)

**Comment by sdroege** [comment]
This is now available as 0.3.1 on crates.io too. Thanks again for reporting!

**Comment by DarkEld3r** [comment]
@sdroege Thanks for the quick response!
